lib/generator.js

```diff
@@ -1297,76 +1297,81 @@
-            visibleInputHeight  = visibleInputBounds.bottom - visibleInputBounds.top,
-            paddedInputWidth    = paddedInputBounds.right   - paddedInputBounds.left,
-            paddedInputHeight   = paddedInputBounds.bottom  - paddedInputBounds.top,
-            
-            // How much of the width is due to effects
-            effectsInputWidth   = visibleInputWidth  - staticInputWidth,
-            effectsInputHeight  = visibleInputHeight - staticInputHeight,
-            // How much of the width is due to padding (mask)
-            paddingInputWidth   = paddedInputWidth  - visibleInputWidth,
-            paddingInputHeight  = paddedInputHeight - visibleInputHeight,
-
-            // Designated image size
-            paddedOutputWidthFloat = targetWidth  || (paddedInputWidth *
-                                    (targetHeight  ? (targetHeight  / paddedInputHeight) : targetScaleX)),
-            paddedOutputHeightFloat = targetHeight || (paddedInputHeight *
-                                    (targetWidth ? (targetWidth / paddedInputWidth) : targetScaleY)),
-
-            // Effects are not scaled when the transformation is non-uniform
-            paddedFloatRatioDiff = Math.abs(paddedOutputWidthFloat / paddedInputWidth -
-                                            paddedOutputHeightFloat / paddedInputHeight),
-            floatDiffEpislon = 2e-16,
-            effectsScaled       = (targetWidth || targetHeight) ? paddedFloatRatioDiff < floatDiffEpislon :
-                                                                  targetScaleX === targetScaleY,
-
-            paddedOutputWidth   = Math.round(paddedOutputWidthFloat),
-            paddedOutputHeight  = Math.round(paddedOutputHeightFloat),
-
-            paddedOutputScaleX  = paddedOutputWidthFloat  / paddedInputWidth,
-            paddedOutputScaleY  = paddedOutputHeightFloat / paddedInputHeight,
-
-            // How much to scale everything that can be scaled (static + padding, maybe effects)
-            scaleX              = effectsScaled ? paddedOutputScaleX : paddedOutputScaleX +
-                                    (effectsInputWidth  * (paddedOutputScaleX - 1)) /
-                                    (staticInputWidth  + paddingInputWidth),
-            scaleY              = effectsScaled ? paddedOutputScaleY : paddedOutputScaleY +
-                                    (effectsInputHeight * (paddedOutputScaleY - 1)) /
-                                    (staticInputHeight + paddingInputHeight),
-
-            // The expected size of the pixmap returned by Photoshop (does not include padding)
-            visibleOutputWidth  = effectsScaled ? scaleX * visibleInputWidth :
-                                                        scaleX * staticInputWidth + effectsInputWidth,
-            visibleOutputHeight = effectsScaled ? scaleY * visibleInputHeight :
+            var staticInputWidth    = staticInputBounds.right   - staticInputBounds.left,
```

Comment:
You are correct. the only changes are that since we know  (targetWidth || targetHeight) is true we can remove some redundant checks from this block of code